library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -382,8 +386,16 @@ private void download(FileDownloadInfo info) {
         executor.submit(downloadThread);
     }
 
-    private void updateTotalBytesFor(Collection<FileDownloadInfo> downloadInfos) {
-        ContentValues values = new ContentValues();
+    private long lastUpdate;
+
+    private void updateTotalBytesIfNecessaryFor(Collection<FileDownloadInfo> downloadInfos) {
+        if (SystemClock.elapsedRealtime() - lastUpdate < ONE_SECOND) {
```

Comment:
I've end up doing this https://github.com/novoda/download-manager/commit/d8ca9569957b24577dfaa8c4da5295505a4946ac so everything is encapsulated :)